src/algebra/monoid_algebra.lean

```diff
@@ -336,7 +344,15 @@ end algebra
 
 section lift
 
-variables {k G} [comm_semiring k] [monoid G] {A : Type u₃} [semiring A] [algebra k A]
+variables {k G} [comm_semiring k] [monoid G]
+variables {A : Type u₃} [semiring A] [algebra k A] {B : Type*} [semiring B] [algebra k B]
+
+/-- `lift_nc_ring_hom` as a `alg_hom`, for when `f` is an `alg_hom` -/
+def lift_nc_alg_hom (f : A →ₐ[k] B) (g : G →* B) (h_comm : ∀ x y, commute (f x) (g y)) :
+  monoid_algebra A G →ₐ[k] B :=
+{ to_fun := lift_nc_ring_hom (f : A →+* B) g h_comm,
+  commutes' := by simp [lift_nc_ring_hom],
+  ..(lift_nc_ring_hom (f : A →+* B) g h_comm)}
```

Comment:
Would this be better or worse as:
```suggestion
{ to_fun := lift_nc ((f : A →+* B) : A →+ B) g,
  commutes' := by simp,
  ..(lift_nc_ring_hom (f : A →+* B) g h_comm)}
```

Reply:
I don't think that this is important: we shouldn't unfold `lift` anyway.

Reply:
Should I remove the `to_fun` entirely then and let the `..` populate it?

Reply:
The only difference is what you'll get with `simp [lift_nc_alg_hom]`. With `to_fun` it will be simplified to `lift_nc`. Without `to_fun` it will be simplified to the definition of `lift_nc`.